Results Rewrite - Generative Model (#59)
greenelab/text_mined_hetnet_manuscript: content/04.results.md

### Generative Model Using Randomly Sampled Label Functions

Creating label functions is a labor-intensive process that can take days to accomplish.
We sought to accelerate this process by measuring how well label functions can be reused.

> We evaluated this notion by performing an experiment where label functions are sampled on an individual (edge vs. edge) level and a global (collective pool of sources) level.

Reviewer: ```suggestion
We evaluated this by performing an experiment where label functions are sampled on an individual (edge vs. edge) level and a global (collective pool of sources) level.
```

We observed that performance increased when edge-specific label functions were added to an edge-specific baseline model, while label function reuse usually provided less benefit (AUROC Figure {@fig:auroc_gen_model_test_set}, AUPR Supplemental Figure {@fig:aupr_gen_model_test_set}).
The quintessential example of this overarching trend is the Compound treats Disease (CtD) edge type, where edge-specific label functions consistently outperformed transferred label functions.

> However, there is evidence that label function transferability is possible for selected edge types and label function sources. 

Reviewer: ```suggestion
However, there is evidence that label function transferability may be feasible for selected edge types and label function sources. 
```

Performance increases as more GiG label functions are incorporated into the CbG baseline model and vice versa.

> This trend suggests that sentences for GiG and CbG may share similar linguistic features or terminology that allows for label functions to be reused.

Reviewer: ```suggestion
This trend suggests that sentences for GiG and CbG may share similar linguistic features or terminology that allows for label functions to be reused, which could relate to both describing physical interaction relationships.
```
